mkosi/__init__.py

```diff
@@ -1126,28 +1126,39 @@ def build_uki(
     run(cmd)
 
 
+def want_uki(config: MkosiConfig) -> bool:
+    # Do we want to build an UKI according to config?
+    # Note that this returns True also in the case where autodetection might later
+    # cause the UKI not to be installed after the file system has been populated.
+
+    if config.bootable == ConfigFeature.disabled:
+        return False
+
+    if config.bootloader == Bootloader.none:
+        return False
+
+    if (config.output_format in (OutputFormat.cpio, OutputFormat.uki) and
+        config.bootable == ConfigFeature.auto):
+        return False
+
+    if (config.architecture.to_efi() is None and
+        config.bootable == ConfigFeature.auto):
```

Comment:
Nit: both are still in the line length limit
```suggestion
    if config.output_format in (OutputFormat.cpio, OutputFormat.uki) and config.bootable == ConfigFeature.auto:
        return False

    if config.architecture.to_efi() is None and config.bootable == ConfigFeature.auto:
```

Other nit: `config.output_format in (OutputFormat.cpio, OutputFormat.uki)` comes up a few more times and on `Architecture`  we have a few helper predicates that say what about this it is, we are interested in. Something like that would be nice on `OutputFormat`, too.